lib/doctest.js

```diff
@@ -10,63 +10,57 @@
 
 'use strict';
 
-var fs = require ('fs');
-var pathlib = require ('path');
+const fs = require ('fs');
+const pathlib = require ('path');
 
-var CoffeeScript = require ('coffeescript');
-var esprima = require ('esprima');
-var _show = require ('sanctuary-show');
-var Z = require ('sanctuary-type-classes');
+const CoffeeScript = require ('coffeescript');
+const esprima = require ('esprima');
+const _show = require ('sanctuary-show');
+const Z = require ('sanctuary-type-classes');
 
-var common = require ('./common');
 
-
-function inferType(path) {
+const inferType = path => {
   switch (pathlib.extname (path)) {
     case '.coffee': return 'coffee';
     case '.js':     return 'js';
     default:        return null;
   }
-}
-
-var rewriters = {coffee: rewrite$coffee, js: rewrite$js};
+};
 
-function evaluate(moduleType, source, path) {
-  return moduleType === 'commonjs' ?
+const evaluate = (moduleType, source, path) =>
+  moduleType === 'commonjs' ?
     commonjsEval (source, path) :
     functionEval (source);
-}
 
-module.exports = function(path, options) {
+module.exports = (path, options) => {
   if (options.module != null &&
       options.module !== 'amd' &&
       options.module !== 'commonjs') {
-    return Promise.reject (new Error (
-      'Invalid module `' + options.module + "'"
-    ));
+    return Promise.reject (new Error (`Invalid module \`${options.module}'`));
   }
   if (options.type != null &&
       options.type !== 'coffee' &&
       options.type !== 'js') {
-    return Promise.reject (new Error (
-      'Invalid type `' + options.type + "'"
-    ));
+    return Promise.reject (new Error (`Invalid type \`${options.type}'`));
   }
 
-  var type = options.type == null ? inferType (path) : options.type;
+  const type = options.type == null ? inferType (path) : options.type;
   if (type == null) {
     return Promise.reject (new Error (
       'Cannot infer type from extension'
     ));
   }
 
-  var source = toModule (
+  const rewriters = {coffee: rewrite$coffee, js: rewrite$js};
+  const source = toModule (
     rewriters[type] (
       {prefix: options.prefix == null ? '' : options.prefix,
        openingDelimiter: options.openingDelimiter,
        closingDelimiter: options.closingDelimiter,
        sourceType: 'script'},
-      common.sanitizeFileContents (fs.readFileSync (path, 'utf8'))
+      (fs.readFileSync (path, 'utf8')
+       .replace (/\r\n?/g, '\n')
+       .replace (/^#!.*/, ''))
```

Comment:
I decided to duplicate this small amount of code to allow __lib/common.js__ to be deleted. Template strings obviate the need for `unlines`, so `sanitizeFileContents` was the only remaining shared function.